cmd/libsnap-confine-private/cleanup-funcs.c

```diff
@@ -28,6 +28,14 @@ void sc_cleanup_string(char **ptr)
 	}
 }
 
+void sc_cleanup_shallow_strv(const char ***ptr)
```

Comment:
Please add a minimal test for this in cleanup-funcs-test.c. Those rely on valgrind mostly something as simple as:

```
static void test_cleanup_shallow_strv(void)
{
	/* It is safe to use with a NULL pointer */
	sc_cleanup_shallow_strv(NULL);

	char **argses = NULL;
	/* It is ok of the pointer value is NULL */
	sc_cleanup_shallow_strv(&argses);
	g_assert_null(argses);

	argses = calloc(10, sizeof char*);
	g_assert_nonnull(argses);
	/* Fill with bogus pointers so attempts to free them would segfault */
	for (int i = 0; i < 10; i++) {
		argses[i] = 0x100 + i;
	}
	sc_cleanup_shallow_strv(&argses);
	g_assert_null(argses);
	/* If we are alive at this point, most likely only the array was free'd */
}

...
static void __attribute__((constructor)) init(void)
{
	...
	g_test_add_func("/cleanup/shallow_strv", test_cleanup_shallow_strv);
}

```